ocp_resources/multi_network_policy.py

```diff
@@ -0,0 +1,92 @@
+from ocp_resources.resource import NamespacedResource
+
+
+IP_BLOCK = "ipBlock"
+
+
+class MultiNetworkPolicy(NamespacedResource):
+    """
+    MultiNetworkPolicy object.
+
+    API reference:
+    https://docs.openshift.com/container-platform/4.14/networking/multiple_networks/configuring-multi-network-policy.html
+    """
+
+    api_group = NamespacedResource.ApiGroup.K8S_CNI_CNCF_IO
+    # api_version = f"{NamespacedResource.ApiGroup.K8S_CNI_CNCF_IO}/{NamespacedResource.ApiVersion.V1BETA1}"
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        nad_name_for_policy=None,
+        client=None,
+        policy_types=None,
+        ingress=None,
+        egress=None,
+        pod_selector=None,
+        *args,
+        **kwargs,
+    ):
+        """
+        Create and manage MultiNetworkPolicy
+
+        Args:
+            name (str): Name of the MultiNetworkPolicy.
+            namespace (str): The namespace where the affected network is located.
+            nad_name_for_policy (str): The name of the NetworkAttachmentDefinition to which the resources connected
+                the created policy will impact.
+            client (DynamicClient): Dynamic client for connecting to a remote cluster.
+            policy_types (list): one or both of the valid ip policies - "Ingress" and "Egress".
+            ingress (list): list of a dict specifying the allowed "from" parameters. if an "Ingress" policy type is
+                required, an ingress list must be provided.
+            egress (list): list of a dict specifying the allowed "to" parameters. if an "Ingress" policy type is
+                required, an egress list must be provided.
+            pod_selector (dict): Map a label to match to. This field is
+        """
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            *args,
+            **kwargs,
+        )
+        self.nad_name_for_policy = nad_name_for_policy
+        self.policy_types = policy_types
+        self.pod_selector = pod_selector
+        self.ingress = ingress
+        self.egress = egress
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            if not self.nad_name_for_policy:
+                raise ValueError("nad_name_for_policy is required")
+            if not self.policy_types:
+                raise ValueError("policy_types is required")
```

Comment:
according to `Allow ingress from all pods in the same namespace` section, this is not mandatory